### PR TITLE
Sync validation of SenderId against authenticated user

### PR DESCRIPTION
**Pull request by Mech0z**
## References

* #1230

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.FunctionHost/Charges/ChargeIngestion.cs [hunk 1]
```diff
@@ -67,6 +84,14 @@ public class ChargeIngestion
             return _httpResponseBuilder.CreateAcceptedResponse(req);
         }
 
+        private bool AuthenticatedMatchesSenderId(SchemaValidatedInboundMessage<ChargeCommandBundle> inboundMessage)
+        {
+            var authorizedActor = _actorContext.CurrentActor;
+            var senderId = inboundMessage.ValidatedMessage?.ChargeCommands.First().Document.Sender.Id;
+
+            return authorizedActor is null || senderId == authorizedActor.Identifier;
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.FunctionHost/Charges/ChargeIngestion.cs:92]: If `authorizedActor is null` is that not an error?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure.Core/Function/HttpResponseBuilder.cs [hunk 2]
```diff
@@ -50,5 +50,12 @@ private void AddHeaders(HttpResponseData httpResponseData)
         {
             httpResponseData.Headers.Add("CorrelationId", _correlationContext.Id);
         }
+
+        public HttpResponseData CreateBadRequestWithErrorText(HttpRequestData request, string errorText)
+        {
+            var httpResponse = request.CreateResponse(HttpStatusCode.BadRequest);
+            httpResponse.WriteString(errorText);
+            return httpResponse;
+        }
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure.Core/Function/HttpResponseBuilder.cs:59]: `public CreateBadRequestWithErrorText` Should be placed before `private AddHeaders`

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.FunctionHost/Charges/ChargeIngestion.cs (hunk 1 above)
Comment (x-platformcoder) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.FunctionHost/Charges/ChargeIngestion.cs:92]: "is null" er indført i C#7, negeret "is not null" er indført i C#9 https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/patterns#constant-pattern, men om der er nogen inhouse regel der gør man skal bruge == og != ved tjek på null så er den god nok  :-)

Comment (HenrikSommer) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.FunctionHost/Charges/ChargeIngestion.cs:92]: My question was not whether to use `is null` or `== null` but rather that if `authorizedActor` is null, I think the method should return false.

Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.FunctionHost/Charges/ChargeIngestion.cs]: Changed to fail if actor is null

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure.Core/Function/HttpResponseBuilder.cs (hunk 2 above)
Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure.Core/Function/HttpResponseBuilder.cs:54]: Done, also gave build errors

**Review by x-platformcoder — APPROVED** [review]
ok

**Pull request merged**